File: data/src/main/java/run/piece/dev/data/model/MemberDTO.java
package run.piece.dev.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * packageName    : com.bsstandard.piece.data.dto
 * fileName       : MemberDTO
 * author         : piecejhm
 * date           : 2022/09/03
 * description    : 회원 정보 조회시 Response DTO
 * ===========================================================
 * DATE              AUTHOR             NOTE
 * -----------------------------------------------------------
 * 2022/09/03        piecejhm       최초 생성
 */
public class MemberDTO {
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("message")
    @Expose
    private Object message;
    @SerializedName("data")
    @Expose
    private Data data;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Object getMessage() {
        return message;
    }

    public void setMessage(Object message) {
        this.message = message;
    }

    public Data getData() {
        return data;
    }

    public void setData(Data data) {
        this.data = data;
    }

    public class Data {

        @SerializedName("memberId")
        @Expose
        private String memberId;
        @SerializedName("name")
        @Expose
        private String name;
        @SerializedName("pinNumber")
        @Expose
        private Object pinNumber;
        @SerializedName("cellPhoneNo")
        @Expose
        private String cellPhoneNo;
        @SerializedName("cellPhoneIdNo")
        @Expose
        private String cellPhoneIdNo;
        @SerializedName("birthDay")
        @Expose
        private String birthDay;
        @SerializedName("zipCode")
        @Expose
        private String zipCode;
        @SerializedName("baseAddress")
        @Expose
        private Object baseAddress;
        @SerializedName("detailAddress")
        @Expose
        private Object detailAddress;
        @SerializedName("idNo")
        @Expose
        private Object idNo;
        @SerializedName("ci")
        @Expose
        private String ci;
        @SerializedName("di")
        @Expose
        private String di;
        @SerializedName("gender")
        @Expose
        private String gender;
        @SerializedName("email")
        @Expose
        private String email;
        @SerializedName("isFido")
        @Expose
        private String isFido;
        @SerializedName("isIdNo")
        @Expose
        private String isIdNo;
        @SerializedName("createdAt")
        @Expose
        private String createdAt;
        @SerializedName("joinDay")
        @Expose
        private String joinDay;

        @SerializedName("totalProfitAmount")
        @Expose
        private String totalProfitAmount;

        @SerializedName("requiredConsentDate")
        @Expose
        private String requiredConsentDate;

        @SerializedName("notRequiredConsentDate")
        @Expose
        private String notRequiredConsentDate;

        @SerializedName("vran")
        @Expose
        private String vran;

        @SerializedName("invstDepsBal") // 나의 예치금 잔액
        @Expose
        private String invstDepsBal;

        @SerializedName("rtnAblAmt") // 출금 가능 잔액
        @Expose
        private String rtnAblAmt;

        /** 블록체인 ID 배포 승인 전이라 주석 처리 **/
//        @SerializedName("signerId")
//        @Expose
//        private String signerId;
        /******************************************/
        @SerializedName("notification")
        @Expose
        private Notification notification;
        @SerializedName("device")
        @Expose
        private Device device;
        @SerializedName("consents")
        @Expose
        private List<Consent> consents = null;
        @SerializedName("bookmarks")
        @Expose
        private List<Bookmark> bookmarks = null;
        @SerializedName("portfolioNotifications")
        @Expose
        private List<Object> portfolioNotifications = null;

        public String getMemberId() {
            return memberId;
        }

        public void setMemberId(String memberId) {
            this.memberId = memberId;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public Object getPinNumber() {
            return pinNumber;
        }

        public void setPinNumber(Object pinNumber) {
            this.pinNumber = pinNumber;
        }

        public String getCellPhoneNo() {
            return cellPhoneNo;
        }

        public void setCellPhoneNo(String cellPhoneNo) {
            this.cellPhoneNo = cellPhoneNo;
        }

        public String getCellPhoneIdNo() {
            return cellPhoneIdNo;
        }

        public void setCellPhoneIdNo(String cellPhoneIdNo) {
            this.cellPhoneIdNo = cellPhoneIdNo;
        }

        public String getBirthDay() {
            return birthDay;
        }

        public void setBirthDay(String birthDay) {
            this.birthDay = birthDay;
        }

        public String getZipCode() {
            return zipCode;
        }

        public void setZipCode(String zipCode) {
            this.zipCode = zipCode;
        }

        public Object getBaseAddress() {
            return baseAddress;
        }

        public void setBaseAddress(Object baseAddress) {
            this.baseAddress = baseAddress;
        }

        public Object getDetailAddress() {
            return detailAddress;
        }

        public void setDetailAddress(Object detailAddress) {
            this.detailAddress = detailAddress;
        }

        public Object getIdNo() {
            return idNo;
        }

        public void setIdNo(Object idNo) {
            this.idNo = idNo;
        }

        public String getCi() {
            return ci;
        }

        public void setCi(String ci) {
            this.ci = ci;
        }

        public String getDi() {
            return di;
        }

        public void setDi(String di) {
            this.di = di;
        }

        public String getGender() {
            return gender;
        }

        public void setGender(String gender) {
            this.gender = gender;
        }

        public String getEmail() {
            return email;
        }

        public void setEmail(String email) {
            this.email = email;
        }

        public String getIsFido() {
            return isFido;
        }

        public void setIsFido(String isFido) {
            this.isFido = isFido;
        }

        public String getIsIdNo() {
            return isIdNo;
        }

        public void setIsIdNo(String isIdNo) {
            this.isIdNo = isIdNo;
        }

        public String getCreatedAt() {
            return createdAt;
        }

        public void setCreatedAt(String createdAt) {
            this.createdAt = createdAt;
        }

        public String getJoinDay() {
            return joinDay;
        }

        public void setJoinDay(String joinDay) {
            this.joinDay = joinDay;
        }

        public String getTotalProfitAmount() {
            return totalProfitAmount;
        }

        public void setTotalProfitAmount(String totalProfitAmount) {
            this.totalProfitAmount = totalProfitAmount;
        }

        public String getRequiredConsentDate() {
            return requiredConsentDate;
        }

        public void setRequiredConsentDate(String requiredConsentDate) {
            this.requiredConsentDate = requiredConsentDate;
        }

        public String getNotRequiredConsentDate() {
            return notRequiredConsentDate;
        }

        public void setNotRequiredConsentDate(String notRequiredConsentDate) {
            this.notRequiredConsentDate = notRequiredConsentDate;
        }

        public String getVran() {
            return vran;
        }

        public void setVran(String vran) {
            this.vran = vran;
        }

        public String getInvstDepsBal() {
            return invstDepsBal;
        }

        public void setInvstDepsBal(String invstDepsBal) {
            this.invstDepsBal = invstDepsBal;
        }

        public String getRtnAblAmt() {
            return rtnAblAmt;
        }

        public void setRtnAblAmt(String rtnAblAmt) {
            this.rtnAblAmt = rtnAblAmt;
        }

        /** 블록체인 ID 배포 승인 전이라 주석 처리 **/
//        public String getSignerId() {
//            return signerId;
//        }
//
//        public void setSignerId(String signerId) {
//            this.signerId = signerId;
//        }
        /************************************/

        public Notification getNotification() {
            return notification;
        }

        public void setNotification(Notification notification) {
            this.notification = notification;
        }

        public Device getDevice() {
            return device;
        }

        public void setDevice(Device device) {
            this.device = device;
        }

        public List<Consent> getConsents() {
            return consents;
        }

        public void setConsents(List<Consent> consents) {
            this.consents = consents;
        }

        public List<Bookmark> getBookmarks() {
            return bookmarks;
        }

        public void setBookmarks(List<Bookmark> bookmarks) {
            this.bookmarks = bookmarks;
        }

        public List<Object> getPortfolioNotifications() {
            return portfolioNotifications;
        }

        public void setPortfolioNotifications(List<Object> portfolioNotifications) {
            this.portfolioNotifications = portfolioNotifications;
        }


        public class Notification {

            @SerializedName("memberId")
            @Expose
            private String memberId;
            @SerializedName("assetNotification")
            @Expose
            private String assetNotification;
            @SerializedName("portfolioNotification")
            @Expose
            private String portfolioNotification;
            @SerializedName("marketingSms")
            @Expose
            private String marketingSms;
            @SerializedName("marketingApp")
            @Expose
            private String marketingApp;

            @SerializedName("isAd")
            @Expose
            private String isAd;

            @SerializedName("isNotice")
            @Expose
            private String isNotice;

            public String getMemberId() {
                return memberId;
            }

            public void setMemberId(String memberId) {
                this.memberId = memberId;
            }

            public String getAssetNotification() {
                return assetNotification;
            }

            public void setAssetNotification(String assetNotification) {
                this.assetNotification = assetNotification;
            }

            public String getPortfolioNotification() {
                return portfolioNotification;
            }

            public void setPortfolioNotification(String portfolioNotification) {
                this.portfolioNotification = portfolioNotification;
            }

            public String getMarketingSms() {
                return marketingSms;
            }

            public void setMarketingSms(String marketingSms) {
                this.marketingSms = marketingSms;
            }

            public String getMarketingApp() {
                return marketingApp;
            }

            public void setMarketingApp(String marketingApp) {
                this.marketingApp = marketingApp;
            }

            public String getIsAd() {
                return isAd;
            }

            public void setIsAd(String isAd) {
                this.isAd = isAd;
            }

            public String getIsNotice() {
                return isNotice;
            }

            public void setIsNotice(String isNotice) {
                this.isNotice = isNotice;
            }
        }

        public class Device {

            @SerializedName("memberId")
            @Expose
            private String memberId;
            @SerializedName("deviceId")
            @Expose
            private String deviceId;
            @SerializedName("deviceOs")
            @Expose
            private String deviceOs;
            @SerializedName("deviceState")
            @Expose
            private String deviceState;
            @SerializedName("fcmToken")
            @Expose
            private String fcmToken;
            @SerializedName("fbToken")
            @Expose
            private String fbToken;

            public String getMemberId() {
                return memberId;
            }

            public void setMemberId(String memberId) {
                this.memberId = memberId;
            }

            public String getDeviceId() {
                return deviceId;
            }

            public void setDeviceId(String deviceId) {
                this.deviceId = deviceId;
            }

            public String getDeviceOs() {
                return deviceOs;
            }

            public void setDeviceOs(String deviceOs) {
                this.deviceOs = deviceOs;
            }

            public String getDeviceState() {
                return deviceState;
            }

            public void setDeviceState(String deviceState) {
                this.deviceState = deviceState;
            }

            public String getFcmToken() {
                return fcmToken;
            }

            public void setFcmToken(String fcmToken) {
                this.fcmToken = fcmToken;
            }

            public String getFbToken() {
                return fbToken;
            }

            public void setFbToken(String fbToken) {
                this.fbToken = fbToken;
            }

        }



        public class Consent {

            @SerializedName("memberId")
            @Expose
            private String memberId;
            @SerializedName("consentCode")
            @Expose
            private String consentCode;
            @SerializedName("isAgreement")
            @Expose
            private String isAgreement;

            public String getMemberId() {
                return memberId;
            }

            public void setMemberId(String memberId) {
                this.memberId = memberId;
            }

            public String getConsentCode() {
                return consentCode;
            }

            public void setConsentCode(String consentCode) {
                this.consentCode = consentCode;
            }

            public String getIsAgreement() {
                return isAgreement;
            }

            public void setIsAgreement(String isAgreement) {
                this.isAgreement = isAgreement;
            }

        }

        public class Bookmark {

            @SerializedName("memberId")
            @Expose
            private String memberId;
            @SerializedName("magazineId")
            @Expose
            private String magazineId;
            @SerializedName("magazineType")
            @Expose
            private String magazineType;
            @SerializedName("title")
            @Expose
            private String title;
            @SerializedName("midTitle")
            @Expose
            private String midTitle;
            @SerializedName("smallTitle")
            @Expose
            private String smallTitle;
            @SerializedName("author")
            @Expose
            private String author;
            @SerializedName("representThumbnailPath")
            @Expose
            private String representThumbnailPath;
            @SerializedName("representImagePath")
            @Expose
            private String representImagePath;
            @SerializedName("contents")
            @Expose
            private String contents;
            @SerializedName("isDelete")
            @Expose
            private String isDelete;
            @SerializedName("createdAt")
            @Expose
            private String createdAt;
            @SerializedName("isFavorite")
            @Expose
            private String isFavorite;

            public String getMemberId() {
                return memberId;
            }

            public void setMemberId(String memberId) {
                this.memberId = memberId;
            }

            public String getMagazineId() {
                return magazineId;
            }

            public void setMagazineId(String magazineId) {
                this.magazineId = magazineId;
            }

            public String getMagazineType() {
                return magazineType;
            }

            public void setMagazineType(String magazineType) {
                this.magazineType = magazineType;
            }

            public String getTitle() {
                return title;
            }

            public void setTitle(String title) {
                this.title = title;
            }

            public String getMidTitle() {
                return midTitle;
            }

            public void setMidTitle(String midTitle) {
                this.midTitle = midTitle;
            }

            public String getSmallTitle() {
                return smallTitle;
            }

            public void setSmallTitle(String smallTitle) {
                this.smallTitle = smallTitle;
            }

            public String getAuthor() {
                return author;
            }

            public void setAuthor(String author) {
                this.author = author;
            }

            public String getRepresentThumbnailPath() {
                return representThumbnailPath;
            }

            public void setRepresentThumbnailPath(String representThumbnailPath) {
                this.representThumbnailPath = representThumbnailPath;
            }

            public String getRepresentImagePath() {
                return representImagePath;
            }

            public void setRepresentImagePath(String representImagePath) {
                this.representImagePath = representImagePath;
            }

            public String getContents() {
                return contents;
            }

            public void setContents(String contents) {
                this.contents = contents;
            }

            public String getIsDelete() {
                return isDelete;
            }

            public void setIsDelete(String isDelete) {
                this.isDelete = isDelete;
            }

            public String getCreatedAt() {
                return createdAt;
            }

            public void setCreatedAt(String createdAt) {
                this.createdAt = createdAt;
            }

            public String getIsFavorite() {
                return isFavorite;
            }

            public void setIsFavorite(String isFavorite) {
                this.isFavorite = isFavorite;
            }

        }
    }

}

File: data/src/main/java/run/piece/dev/data/refactoring/ui/member/model/dto/MemberDto.kt
package run.piece.dev.data.refactoring.ui.member.model.dto

import com.google.gson.annotations.SerializedName

class MemberDto(@SerializedName("memberId") var memberId: String?,
                @SerializedName("name") var name: String?,
                @SerializedName("pinNumber") var pinNumber: String?,
                @SerializedName("cellPhoneNo") var cellPhoneNo: String?,
                @SerializedName("cellPhoneIdNo") var cellPhoneIdNo: String?,
                @SerializedName("birthDay") var birthDay: String?,
                @SerializedName("zipCode") var zipCode: String?,
                @SerializedName("baseAddress") var baseAddress: String?,
                @SerializedName("detailAddress") var detailAddress: String?,
                @SerializedName("idNo") var idNo: Int?,
                @SerializedName("ci") var ci: String?,
                @SerializedName("di") var di: String?,
                @SerializedName("gender") var gender: String?,
                @SerializedName("email") var email: String?,
                @SerializedName("isFido") var isFido: String?,
                @SerializedName("ssn") var ssn: String?,
                @SerializedName("isIdNo") var isIdNo: String?,
                @SerializedName("createdAt") var createdAt: String?,
                @SerializedName("joinDay") var joinDay: String?,
                @SerializedName("totalProfitAmount") var totalProfitAmount: String?,
                @SerializedName("requiredConsentDate") var requiredConsentDate: String?,
                @SerializedName("notRequiredConsentDate") var notRequiredConsentDate: String?,
                @SerializedName("vran") var vran: String?,
                @SerializedName("invstDepsBal") var invstDepsBal: String?,
                @SerializedName("rtnAblAmt") var rtnAblAmt: String?,
                @SerializedName("publicKey") var publicKey: String?,
                @SerializedName("notification") var notification: MemberNotificationDto?,
                @SerializedName("device") var device: MemberDeviceDto?,
                @SerializedName("consents") var consents: List<MemberConsentDto>?,
                @SerializedName("bookmarks") var bookmarks: List<MemberBookmarkDto>?,
                @SerializedName("portfolioNotifications") var portfolioNotifications: List<MemberNotificationItemDto>?,
                @SerializedName("preference") var preference: MemberPreferenceDto?,
)

// 회원가입 DTO
class JoinDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("deviceId") var deviceId: String?,
    @SerializedName("accessToken") var accessToken: String?,
    @SerializedName("refreshToken") var refreshToken: String?,
    @SerializedName("expiredAt") var expiredAt: String?
)

// 회원 핀번호 검증 DTO
class AuthPinDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("pinNumber") var pinNumber: String?,
    @SerializedName("passwordUpdatedAt") var passwordUpdatedAt: String?,
    @SerializedName("passwordIncorrectCount") var passwordIncorrectCount: String?,
    @SerializedName("isExistMember") var isExistMember: String?
    )

// 실명인증 여부 조회 DTO
class SsnDto(
    @SerializedName("publicKey") var publicKey: String?,
    @SerializedName("ssnYn") var ssnYn: String?
)

class MemberNotificationDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("assetNotification") var assetNotification: String?,
    @SerializedName("isAd") var isAd: String?,
    @SerializedName("isNotice") var isNotice: String?,
    @SerializedName("marketingApp") var marketingApp: String?,
    @SerializedName("marketingSms") var marketingSms: String?,
    @SerializedName("portfolioNotification") var portfolioNotification: String?
)

class MemberDeviceDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("deviceId") var deviceId: String?,
    @SerializedName("deviceOs") var deviceOs: String?,
    @SerializedName("deviceState") var deviceState: String?,
    @SerializedName("fcmToken") var fcmToken: String?,
    @SerializedName("fbToken") var fbToken: String?
)

class MemberConsentDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("consentCode") var consentCode: String?,
    @SerializedName("isAgreement") var isAgreement: String?
)

class MemberBookmarkDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("magazineId") var magazineId: String?,
    @SerializedName("magazineType") var magazineType: String?,
    @SerializedName("title") var title: String?,
    @SerializedName("midTitle") var midTitle: String?,
    @SerializedName("smallTitle") var smallTitle: String?,
    @SerializedName("author") var author: String?,
    @SerializedName("representThumbnailPath") var representThumbnailPath: String?,
    @SerializedName("representImagePath") var representImagePath: String?,
    @SerializedName("contents") var contents: String?,
    @SerializedName("isDelete") var isDelete: String?,
    @SerializedName("createdAt") var createdAt: String?,
    @SerializedName("isFavorite") var isFavorite: String?
)


class MemberNotificationItemDto(
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("portfolioId") var portfolioId: String?,
    @SerializedName("notificationAt") var notificationAt: String?)

class MemberPreferenceDto(
    @SerializedName("resultId") var resultId: String?,
    @SerializedName("minScore") var minScore: Int?,
    @SerializedName("maxScore") var maxScore: Int?,
    @SerializedName("result") var result: String?,
    @SerializedName("description") var description: String?,
    @SerializedName("resultImagePath") var resultImagePath: String?,
    @SerializedName("interestProductDescription") var interestProductDescription: String?,
    @SerializedName("memberId") var memberId: String?,
    @SerializedName("name") var name: String?,
    @SerializedName("score") var score: Int?,
    @SerializedName("count") var count: Int?,
    @SerializedName("isVulnerableInvestors") var isVulnerableInvestors: String?,
    @SerializedName("createdAt") var createdAt: String?,
)